src/main/java/bio/terra/common/logging/RequestLoggingFilter.java

```diff
@@ -0,0 +1,187 @@
+      responseSize = ((ContentCachingResponseWrapper) response).getContentAsByteArray().length;
+    }
+
+    GenericJson json = new GenericJson();
+    json.setFactory(new JacksonFactory());
+    json.set(
+        "httpRequest",
+        getGoogleHttpRequestObject(request, response, latency, requestSize, responseSize));
+    json.set("requestHeaders", new ServletServerHttpRequest(request).getHeaders());
+
+    try {
+      URI uri = new URI(request.getRequestURI());
+      String message =
+          String.format("%s %s %s", request.getMethod(), uri.getPath(), response.getStatus());
+
+      // Log the message, and include the supplementary JSON as an additional arg.
+      // If GoogleJsonLayout has been loaded, it will merge the JSON into the structured log output
+      // for ingestion by Cloud Logging. If the default logback layout is being used, the JSON
+      // argument will be ignored.
+      log.info(message, json);
+    } catch (URISyntaxException e) {
+      e.printStackTrace();
+    }
+
+    debugLogResponse(response);
+  }
+
+  /**
+   * Logs various details about the inbound request at DEBUG severity.
+   *
+   * <p>Note that this method does not produce any structured JSON for the GoogleJsonFormat to
+   * interpret. The goal is primarily to product human-readable output for local or non-production
+   * debugging.
+   */
+  private void debugLogRequest(HttpServletRequest request) throws IOException {
+    GenericJson json = new GenericJson();
+    json.setFactory(new JacksonFactory());
+    json.put("method", request.getMethod());
+    json.put("uri", request.getRequestURI());
+
+    String queryString = request.getQueryString();
+    if (queryString != null) {
+      json.put("query", queryString);
+    }
+
+    HttpHeaders headers = new ServletServerHttpRequest(request).getHeaders();
+    json.put("headers", headers);
+
+    if (request instanceof ContentCachingRequestWrapper) {
+      ContentCachingRequestWrapper wrapper = (ContentCachingRequestWrapper) request;
+      byte[] buf = wrapper.getContentAsByteArray();
+      int length = Math.min(buf.length, 10000);
+      try {
+        json.put("payload", new String(buf, 0, length, wrapper.getCharacterEncoding()));
+      } catch (UnsupportedEncodingException e) {
+      }
+    }
+    log.debug(json.toPrettyString());
+  }
+
+  /** Logs various details about the response at DEBUG severity. */
+  private void debugLogResponse(HttpServletResponse response) throws IOException {
+    GenericJson json = new GenericJson();
+    json.setFactory(new JacksonFactory());
+    json.put("status", response.getStatus());
+
+    HttpHeaders headers = new ServletServerHttpResponse(response).getHeaders();
+    json.put("headers", headers);
+
+    if (response instanceof ContentCachingResponseWrapper) {
+      ContentCachingResponseWrapper wrapper = (ContentCachingResponseWrapper) response;
+      byte[] buf = wrapper.getContentAsByteArray();
+      int length = Math.min(buf.length, 10000);
```

Comment:
Maybe give the loose constant a name for code doc'n.

Reply:
Done